### ISSUE TITLE
Usage of "long" type in the code 

### ISSUE THREAD

**Issue by lucafedeli88** [ECP-WarpX/WarpX#321]
"long" integer type is used in many places in the code (e.g., for particle index in particle containers).
However, on many machines, "int" and "long" are 32 bit integers, while "long long" is a 64 bit integer.  
If the idea is that selected quantities need to be able to go beyond 2147483647, it should be better to use int64_t....

**Comment by WeiqunZhang** [comment]
As far as I understand, long is mostly used for two reasons, (1) compatibility with PICSAR and python interface, and (2) the size.  For the size, yes, on "many" machines the size of long is 4 bytes.  But I don't think we care about integer overflowing on those *Windows* machines, because there are no supercomputers running Windows.  (Even Mac has the size of 8 for long).

**Comment by ax3l** [comment]
Yep, sizes of `short`, `int`, `long` and `long long` depend on the data model. The official standard/docs: https://en.cppreference.com/w/cpp/language/types
Similar issue in real-word between Linux-OSX-Windows, all on 64bit machines: https://github.com/openPMD/openPMD-api/pull/337

TL;DR: Use `uint64_t` if you want to ensure your range is long enough for particle ids et al.
For example, on MSVC for 64bit Windows, `long` and `int` are only 32bit these days: https://godbolt.org/z/qJQzNq

**Comment by lucafedeli88** [comment]
ok, thanks a lot for the clarification!
